opennlp-tools/src/main/java/opennlp/tools/ml/model/Context.java

```diff
@@ -57,4 +59,19 @@ public Context(int[] outcomePattern, double[] parameters) {
   public double[] getParameters() {
     return parameters;
   }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+    else if (obj instanceof Context) {
+      Context context = (Context) obj;
+      return Arrays.equals(parameters, context.parameters)
+           && Arrays.equals(outcomes, context.outcomes);
+    }
+    else {
+      return false;
+    }
+  }
```

Comment:
Rewrite all of the equals() code in all classes, similar to my comment for EvalParameters.java